src/Configuration/src/KubernetesBase/KubernetesConfigMapProvider.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright 2017 the original author or authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using k8s;
+using k8s.Models;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Rest;
+using System;
+using System.Linq;
+using System.Net;
+
+namespace Steeltoe.Extensions.Configuration.Kubernetes
+{
+    internal class KubernetesConfigMapProvider : ConfigurationProvider, IDisposable
+    {
+        private IKubernetes K8sClient { get; set; }
+
+        private KubernetesConfigSourceSettings Settings { get; set; }
+
+        private Watcher<V1ConfigMap> ConfigMapWatcher { get; set; }
+
+        internal KubernetesConfigMapProvider(IKubernetes kubernetes, KubernetesConfigSourceSettings settings)
+        {
+            if (kubernetes is null)
+            {
+                throw new ArgumentNullException(nameof(kubernetes));
+            }
+
+            if (settings is null)
+            {
+                throw new ArgumentNullException(nameof(settings));
+            }
+
+            K8sClient = kubernetes;
+            Settings = settings;
+        }
+
+        public override void Load()
+        {
+            try
+            {
+                var configMapResponse = K8sClient.ListNamespacedConfigMapWithHttpMessagesAsync(Settings.Namespace ?? "default", fieldSelector: $"metadata.name={Settings.Name}", watch: Settings.Watch).GetAwaiter().GetResult();
+                ProcessData(configMapResponse.Body.Items?.FirstOrDefault());
```

Comment:
This is incorrect, if `watch` is true the result will not be deserialized into a V1List type and must be read through watcher

Reply:
This is a good point, its a leftover from a previous approach that didn't end up making sense, thanks